packages/graphql_server/src/starHistory.ts

```diff
@@ -0,0 +1,195 @@
+ * @param {number} maxRequestAmount - Maximum number of API requests to make to retrieve the star records.
+ * @returns {StarRecord[]} - An array of `StarRecord` objects representing the star records.
+ */
+async function getRepoStarsMap(
+  repo: string,
+  token: string,
+  requestPages: number[],
+  maxRequestAmount: number
+) {
+  const resArray = await Promise.all(
+    requestPages.map((page) => {
+      return getRepoStargazers(repo, token, page)
+    })
+  )
+
+  const starRecordsMap: Map<string, number> = new Map()
+
+  if (requestPages.length < maxRequestAmount) {
+    const starRecordsData: { starred_at: string }[] = []
+    resArray.forEach(({ data }) => {
+      starRecordsData.push(...data)
+    })
+    for (let i = 0; i < starRecordsData.length; ) {
+      starRecordsMap.set(getDateString(starRecordsData[i].starred_at), i + 1)
+      i += Math.floor(starRecordsData.length / maxRequestAmount) || 1
+    }
+  } else {
+    resArray.forEach(({ data }: { data: { starred_at: string }[] }, index) => {
+      if (data.length > 0) {
+        const starRecord = data[0]
+        starRecordsMap.set(
+          getDateString(starRecord.starred_at),
+          DEFAULT_PER_PAGE * (requestPages[index] - 1)
+        )
+      }
+    })
+  }
+
+  const starAmount = await getRepoStargazersCount(repo, token)
+  starRecordsMap.set(getDateString(Date.now()), starAmount)
+
+  const starRecords: StarRecord[] = []
+
+  starRecordsMap.forEach((v, k) => {
+    starRecords.push({
+      date: k,
+      count: v
+    })
+  })
+
+  return starRecords
+}
+
+/** Function generates an array of numbers within a given range
+ * This is needed to only reconstruct the star history from some specific points in the history
+ * And not from looking at every single star recorded
+ * @param {number} from
+ * @param {number} to
+ * @returns Array<number>: An array of numbers within the specified range.
+ */
+function range(from: number, to: number): number[] {
+  const r: number[] = []
+  for (let i = from; i <= to; i++) {
+    r.push(i)
+  }
+  return r
+}
+
+/** Retuns the timestamp of a given date
+ * @param {Date | number | string} t - Date, timestamp or string
+ * @returns Timestamp of the given date
+ */
+function getTimeStampByDate(t: Date | number | string): number {
+  const d = new Date(t)
+
+  return d.getTime()
+}
+
+/** Formats a dat into a string using the specified format
+ * @param {Date | number | string} t
+ * @param {string} format - The format string for the desired date format.
+ * Default is "yyyy/MM/dd hh:mm:ss".
+ * @returns Formatted string
+ */
+function getDateString(t: Date | number | string, format = 'yyyy/MM/dd hh:mm:ss'): string {
+  const d = new Date(getTimeStampByDate(t))
+
+  const year = d.getFullYear()
+  const month = d.getMonth() + 1
+  const date = d.getDate()
+  const hours = d.getHours()
+  const minutes = d.getMinutes()
+  const seconds = d.getSeconds()
+
+  return format
```

Comment:
not all formats supported. Make people aware!?

Reply:
This method always gets the same formatted input so I believe that all necessar formats should be supported